types/eslint/index.d.ts

```diff
@@ -679,11 +669,11 @@ export namespace Rule {
 
     interface CodePath {
         id: string;
+        origin: "program" | "function" | "class-field-initializer" | "class-static-block";
```

Comment:
[Feature] When a union type is referenced in another type, it's generally helpful to extract out the union type:

```suggestion
    type CodePathOrigin = "program" | "function" | "class-field-initializer" | "class-static-block";

    interface CodePath {
        id: string;
        origin: CodePathOrigin;
```

* It's a little friendlier for developers to have an explicit name for it
* If external devs ever need to refer to it, it's easier when named (akin to https://github.com/typescript-eslint/typescript-eslint/issues/7670)